### ISSUE TITLE
ask for compine the custom_stoploss with the adjust_trade_position

### ISSUE THREAD

**Issue by amjad-12** [freqtrade/freqtrade#9794]
Hello I have problem in combine the custom_stoploss with the adjust_trade_position and I want to know why although I think I follow the right things 
thank you in advance
My operating system is

  * Operating system: Ubuntu
  * Python Version: 3.11.4
  * CCXT version: 23.0.1
  * Freqtrade Version: latest

I set this in the first 
```
    stoploss = -2.30
    max_entry_position_adjustment = 2
    max_dca_multiplier = 6
    trailing_stop = False
    # trailing_only_offset_is_reached = False
    # trailing_stop_positive = 0.01
    # trailing_stop_positive_offset = 0.0  # Disabled / not configured
    use_custom_stoploss = True
    custom_info = {}
```

I have this in my adjust trade position 
```
def adjust_trade_position(self, trade: Trade, current_time: datetime,
                              current_rate: float, current_profit: float,
                              min_stake: Optional[float], max_stake: float,
                              current_entry_rate: float, current_exit_rate: float,
                              current_entry_profit: float, current_exit_profit: float,
                              **kwargs) -> Optional[float]:

        filled_entries = trade.select_filled_orders(trade.entry_side)
        count_of_entries = trade.nr_of_successful_entries

        if count_of_entries == 1:
            ob = self.dp.orderbook(trade.pair, 1)
            if 'bids' in ob and len(ob['bids']) > 0:
                best_bid = ob['bids'][0][0]
                best_ask = ob['asks'][0][0]
                stored_data = self.custom_info.get(trade.pair, {})
                initial_best_bid = stored_data.get('initial_best_bid')

                if best_bid < initial_best_bid and best_ask <= initial_best_bid:
                    # This returns first order stake size
                    stake_amount = filled_entries[0].stake_amount
                    # This then calculates current safety order size
                    stake_amount = stake_amount * 2
                    # stake_amount = stake_amount * \
                    #     (1 + (count_of_entries * 0.25))

                    return stake_amount
                else:
                    return None
        elif count_of_entries == 2:
            ob = self.dp.orderbook(trade.pair, 1)
            if 'bids' in ob and len(ob['bids']) > 0:
                best_bid = ob['bids'][0][0]
                best_ask = ob['asks'][0][0]
                stored_data = self.custom_info.get(trade.pair, {})
                second_best_bid = stored_data.get('second_best_bid')

                if best_bid < second_best_bid and best_ask <= second_best_bid:

                    # This returns first order stake size
                    stake_amount = filled_entries[0].stake_amount
                    # This then calculates current safety order size
                    stake_amount = stake_amount * 3
                    # stake_amount = stake_amount * \
                    #     (1 + (count_of_entries * 0.25))

                    return stake_amount
                else:
                    return None
        else:
            return None

````

and I have this in my custom_stoploss 
```

    def custom_stoploss(self, pair: str, trade: 'Trade', current_time: datetime,
                        current_rate: float, current_profit: float, after_fill: bool,
                        **kwargs) -> Optional[float]:

        if trade is not None:
            count_of_entries = trade.nr_of_successful_entries
            if count_of_entries == 3:
                stored_data = self.custom_info.get(pair, {})
                third_best_bid = stored_data.get('third_best_bid')
                return stoploss_from_absolute(third_best_bid - third_best_bid * (1 / 100) , current_rate, is_short=trade.is_short)
```

what I want to happen after the second adjusting I want the stop loss to be under the third price buying in it which it the price I adjust the position it for the second time I want it to be under it by 1% what happened now in telegram I hav messag first the unrealized_profit is -1.08% but then i got -6.43 and I loss -6.43 but I should to lose only -1.08% 

this is the messages I got 

```
❌ Binance (dry): Exited VIDT/BTC (#38)
Profit: -6.43% (loss: -3.86068397 BTC / -183011.863 USD)
Exit Reason: trailing_stop_loss
Direction: Long
Amount: 80365097
Open Rate: 0.00000075 BTC
Exit Rate: 0.0000007 BTC
Duration: 0:19:16 (19.3 min)
```
then I got 

```
❌ Binance (dry): Exiting VIDT/BTC (#38)
Unrealized Profit: -1.08% (loss: -0.6492947 BTC / -30779.166 USD)
Exit Reason: trailing_stop_loss
Direction: Long
Amount: 80365097
Open Rate: 0.00000075 BTC
Current Rate: 0.00000074 BTC
Exit Rate: 0.00000074 BTC
Duration: 0:19:16 (19.3 min)

```

but then my loss I got is -6.43% not -1.08%

**Comment by xmatthias** [comment]
Please get used to filling out the issue template.  Those who frequently omit the template may have their issues closed without comment.
We're asking some very basic questions - by deleting the issue template, you're explicitly choosing not to answer our questions.

Filling out the template helps reduce our workload, and we greatly appreciate that.

---

Without knowing your version, the initial  assumption is that you're on an outdated version (messages were slightly improved in recen versions).

Also - your 1% stoploss only applies after 3 entries - which have conditionals respectively - so it's unclear if that trade even got the stoploss adjusted.

Getting different stoploss values between the messages can happen when using market orders - as the market might not have sufficient capital for your trade - so you're experiencing slippage.

Logs (which were not provided) would probably show what is actually happening.

**Comment by amjad-12** [comment]
thank you so much I update my question and answer the template with my operating system and versions

thank you so much 
but I make the stop_loss is 230% I make big number because I want only the custom_stoploss to work when the nr_of_successful_entries equalt to 3 

and please to make sure I understand what happened is when the stoploss return from the custom_stoploss with 1% is the sell order become market? 
and what happened then it's still sell on many prices to fill my market sell order right ?
and the calculation of the price I selling on it to fill my market order I got this loss -6.43 or what and if it is right why the message with profit -6.43 become before the unrealizedprofit with value -1.08
because if it's right I should to follow new way to minimize my loses 

also can I make stoploss on exchange only on the second adjusting please or not

thank you so so much

**Comment by xmatthias** [comment]
Frist off - `latest` is not a valid freqtrade version.
if it's develop - it's a moving target.
if it's not develop - it's a moving-target, too - though a slower moving target.

---

the number (6% over a 1% stoploss) does make kinda sense, as roughly 5% is the maximum slippage we allow for simulated market orders.

what ordertype is actually used is a different configuration - but as said - logs from around the time of the stop might have the explanation - without them - ig's guessing at best.
